From: Don Skidmore <donald.c.skidmore@intel.com>
Date: Fri, 4 Nov 2016 16:46:16 -0400
Subject: ixgbe: Correct X550 phy ID
Patch-mainline: v4.10-rc1
Git-commit: 5f1c3589b0f0feb2c99b985d3a3bbda0dd790626
References: bsc#1024349 FATE#321240 bsc#1024355 FATE#321241

We were using an old Alpha version of the X550 phy ID.  This was leading
to unnecessary queries of the PHY.  I removed the old ID (which shouldn't
be on any HW) and add the two that are.

Signed-off-by: Don Skidmore <donald.c.skidmore@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c  |    3 ++-
 drivers/net/ethernet/intel/ixgbe/ixgbe_type.h |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
@@ -389,7 +389,8 @@ static enum ixgbe_phy_type ixgbe_get_phy
 	case TN1010_PHY_ID:
 		phy_type = ixgbe_phy_tn;
 		break;
-	case X550_PHY_ID:
+	case X550_PHY_ID2:
+	case X550_PHY_ID3:
 	case X540_PHY_ID:
 		phy_type = ixgbe_phy_aq;
 		break;
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_type.h
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_type.h
@@ -1370,7 +1370,8 @@ struct ixgbe_thermal_sensor_data {
 #define TN1010_PHY_ID    0x00A19410
 #define TNX_FW_REV       0xB
 #define X540_PHY_ID      0x01540200
-#define X550_PHY_ID      0x01540220
+#define X550_PHY_ID2	0x01540223
+#define X550_PHY_ID3	0x01540221
 #define X557_PHY_ID      0x01540240
 #define X557_PHY_ID2	0x01540250
 #define QT2022_PHY_ID    0x0043A400
